lib/rubocop/cop/style/hash_except.rb

```diff
@@ -129,11 +129,15 @@ def semantically_except_method?(send, block)
         end
 
         def included?(negated, body)
-          body.method?('include?') || body.method?('in?') || (negated && body.method?('exclude?'))
+          if negated
+            body.method?('exclude?')
+          else
+            body.method?('include?') || body.method?('in?')
+          end
```

Comment:
The error was here, `body.method?('include?') || body.method?('in?')` were evaluated regardless of `negated` param.

For `reject`/`!include?` combination the following happened in `semantically_except_method?` method (it's above this method):
1. `negated = true` (because of `!` in front of `include?`)
2. `:reject === send.method_name` (so line 123 is evaluated)
3. `included?` returns `true` because `body.method?('include?')` is `true`